### PR TITLE
Fix pdf print layout

### PR DESCRIPTION
**Pull request by izhanchang**
Fix multipage quotes being split between rows per issue #426

### CODE REVIEW

**Comment by DaAwesomeP** [comment]
See https://developer.mozilla.org/en-US/docs/Web/CSS/page-break-inside, `page-break-inside` is legacy. We should use [`break-inside`](https://developer.mozilla.org/en-US/docs/Web/CSS/break-inside) instead it seems.

Also were you able to create an invoice before to replicate the issue? I can't actually get it to break the page inside a row.

File: app/views/invoices/_prettyView_footer.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-  <br>
+<br>
```
Comment (DaAwesomeP) [inline app/views/invoices/_prettyView_footer.html.erb:1]: fix indent

File: app/views/invoices/_prettyView_footer.html.erb [hunk 2]
```diff
@@ -17,7 +17,7 @@
   </table>
 </td>
 <tr><td>
-  <table style="float:right;" class=prettyBorder >
+    <table style="float:right;" class=prettyBorder >
```
Comment (DaAwesomeP) [inline app/views/invoices/_prettyView_footer.html.erb:20]: fix indent

File: app/views/invoices/_prettyView_footer.html.erb [hunk 3]
```diff
@@ -42,11 +42,11 @@
   </table>
 </td></tr>
 <tr><td colspan=2>
-<table class=prettyBorder align=center width=100% >
+    <table class=prettyBorder align=center width=100% >
```
Comment (DaAwesomeP) [inline app/views/invoices/_prettyView_footer.html.erb:45]: fix indent

**Comment by izhanchang** [comment]
Unfortunately using `break-inside: avoid;` doesn't seem to fix the issue. I was able to replicate this issue, but the only way I seemed to be able to solve it was to use `page-break-inside`. I did try using different browsers but came to the same result. 
Chrome break-inside
Firefox page-break
Firefox break-inside

File: app/views/invoices/_prettyView_footer.html.erb (hunk 1 above)
Comment (izhanchang) [reply app/views/invoices/_prettyView_footer.html.erb:1]: fixed

File: app/views/invoices/_prettyView_footer.html.erb (hunk 3 above)
Comment (izhanchang) [reply app/views/invoices/_prettyView_footer.html.erb:45]: fixed

**Comment by DaAwesomeP** [comment]
The PDF generation is done on the server for consistency (not in browser), so they definitely should match. 

OK, then we'll keep `page-break-inside`.

When you fix the indents you should make another commit and push it.

One more edge case to test: If the first row/invoice item is really big/long, does it push it to the next page? I am unsure what the correct solution is, but I think we maybe always want the first item to be on the first page (and break the cell if needed) or it might look weird to have an empty first page with just the header/column name row.

**Comment by timparenti** [comment]
The `wicked_pdf` gem was just upgraded from 1.1.0 (Aug 2016) to 2.1.0 (Jun 2020) in 010647b21e98f29e0d16c219e9de0a7f18cf5be9.  If you're still testing with 1.1.0, it's possible its renderer isn't aware of the new `break-inside` property, as it seems support for it was only just getting started then.  Since 2.x only came out in 2020, it would be best to test with the recently-upgraded gem before merging.

**Comment by izhanchang** [comment]
Oh yes, just needed to update the gem. `break-inside` works now, and this should solve what Perry was referencing as the other test case. If this is what we're looking for I'll create another commit and push it.
image
image

**Comment by DaAwesomeP** [comment]
That looks perfect!

**Comment by izhanchang** [comment]
Reopening pull request on Master
